mdp/hinet/layer.py

```diff
@@ -49,6 +50,23 @@ def __init__(self, nodes, dtype=None):
             input_dim += node.input_dim
             self.node_input_dims[index] = node.input_dim
         output_dim = self._get_output_dim_from_nodes()
+
+        # store which nodes are pretrained up to what phase
+        _pretrained_phase = [node.get_current_train_phase()
+                             for node in nodes]
+        # check if all the nodes are already fully trained
+        train_len = 0
+        for i_node, node in enumerate(nodes):
+            if node.is_trainable():
+                train_len += (len(node._get_train_seq())
+                              - _pretrained_phase[i_node])
```

Comment:
Yes, you are right. pre trained phases are not required. I just wanted to know if the node is still training or not. But I can just check if the node's is_training flag is set. Perhaps, I moved this code from the flow when I was not clear about the distinction between trained/trainable/non-trainable nodes. Will fix this. Thanks.